DEVX_ROADMAP.md

```diff
@@ -0,0 +1,50 @@
+
+## Big picture
+* The gateway component will be deprecated. Clients will now communicate directly with full nodes, and full nodes will act as quorum driver (i.e., accept transactions, form a certificate + certificate effects, return the effects to the client)
+* Child objects are getting a big revamp: they no longer need to be passed as explicit transaction inputs, and rich new collection types (e.g., large hash maps with dynamic lookup) will be possible. More details in the Sui Move section.
+
+## JSON RPC
+* [Gateway deprecation] `sui_syncAccountState` will be deprecated and removed.
+* [Gateway deprecation] Gateway’s `sui_executeTransaction` will be replaced by fullnode’s `sui_executeTransaction` which has an extra parameter that allows users to choose if a method should block or return immediately.
+* Pagination support for the following APIs.
+  * `get_objects_owned_by_address`
+  * `get_objects_owned_by_object`
+  * `get_recent_transactions`
+  * `get_transactions_by_input_object`
+  * `get_transactions_by_mutated_object`
+  * `get_transactions_by_move_function`
+  * `get_transactions_from_addr`
+  * `get_transactions_to_addr`
+  * `get_events_by_transaction`
+  * `get_events_by_transaction_module`
+  * `get_events_by_move_event_struct_name`
+  * `get_events_by_sender`
+  * `get_events_by_recipient`
+  * `get_events_by_object`
+  * `get_events_by_timerange`
+* A `sui_call` API that emulates transaction execution, but does not commit the same to the network, akin to eth_call in Ethereum. This is useful for gas cost prediction and transaction effect simulation.
+* [done] `sui_eventSubscription` API that enables subscribing to all transaction events, including ones that don’t generate network activity ([issue](https://github.com/MystenLabs/sui/pull/4288))
+
+## SDK (Typescript, Rust)
+* Reliability improvements in Typescript SDK support for `sui_publish` API
+* Support for `sui_subscribeEvent` in Typescript SDK
+* Rust SDK will be kept at parity with the Typescript SDK when feasible
+
+# Sui Move
+* Support for dynamic access of child objects to nullify the need to explicitly pass child objects in a transaction as well as to enable implementation of design patterns/collections that require pointer-like parent->child representation. ([issue](https://github.com/MystenLabs/sui/issues/4203))
+* Support for passing strings and object ID's encodings to entry functions
+* [done] This frequently requested feature allows passing a vector of objects to entry functions which will open up a variety of flexible use cases wrt payments and coin management.
```

Comment:
@sblackshear did you add the [done]? if they are only merged and not yet tagged for release they strictly speaking it is not usable. It may be good to leave out the [done] notation and direct people to the release notes instead.

Reply:
+1 to Janet